custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -629,3 +662,180 @@ def get_queue(self):
+                self.data["accounting"][addr] = {
+                   "address": arp_hosts[addr]['address'],
+                   "mac-address": arp_hosts[addr]['address']
+                }
+
+        # Build name for host. First try getting DHCP lease comment, then entry in DNS (only static entries)
+        #   and then device's host-name. If everything fails use hosts IP address as name
+        dns_data = parse_api(
+            data={},
+            source=self.api.path("/ip/dns/static", return_list=True),
+            key="address",
+            vals=[
+                {"name": "address"},
+                {"name": "name"},
+            ],
+        )
+
+        for addr in self.data["accounting"]:
+            if str(self.data["accounting"][addr].get('comment', '').strip()):
+                self.data["accounting"][addr]['name'] = self.data["accounting"][addr]['comment']
+            elif addr in dns_data and str(dns_data[addr].get('name', '').strip()):
+                self.data["accounting"][addr]['name'] = dns_data[addr]['name']
+            elif str(self.data["accounting"][addr].get('host-name', '').strip()):
+                self.data["accounting"][addr]['name'] = self.data["accounting"][addr]['host-name']
+            else:
+                self.data["accounting"][addr]['name'] = self.data["accounting"][addr]['address']
+
+        _LOGGER.debug(f"Generated {len(self.data['accounting'])} accounting devices")
+
+        # Build list of local networks
+        dhcp_networks = parse_api(
+            data={},
+            source=self.api.path("/ip/dhcp-server/network", return_list=True),
+            key="address",
+            vals=[
+                {"name": "address"},
+            ],
+            ensure_vals=[
+                {"name": "address"},
+            ]
+        )
+
+        self.local_dhcp_networks = [ipaddress.IPv4Network(network) for network in dhcp_networks]
+
+    def _address_part_of_local_network(self, address):
+        address = ipaddress.ip_address(address)
+        for network in self.local_dhcp_networks:
+            if address in network:
+                return True
+        return False
+
+    def get_accounting(self):
+        """Get Accounting data from Mikrotik"""
+        traffic_type, traffic_div = self._get_traffic_type_and_div()
+
+        # Build temp accounting values dict with all known addresses
+        # Also set traffic type for each item
+        tmp_accounting_values = {}
+        for addr in self.data['accounting']:
+            tmp_accounting_values[addr] = {
+                "wan-tx": 0,
+                "wan-rx": 0,
+                "lan-tx": 0,
+                "lan-rx": 0
+            }
+            self.data['accounting'][addr]["tx-rx-attr"] = traffic_type
+
+        time_diff = self.api.take_accounting_snapshot()
+        if time_diff:
+            accounting_data = parse_api(
+                data={},
+                source=self.api.path("/ip/accounting/snapshot", return_list=True),
```

Comment:
return_list=True is now default. Since that used the most. only return_list=False have to be specified